rfd/0056-buildbox-image-handling.md

```diff
@@ -0,0 +1,74 @@
+---
+authors: Trent Clarke (trent@goteleport.com)
+state: draft
+---
+
+# RFD 45 - Buidlbox image Handling 
+
+## What
+
+A "buildbox" image contains the tools environment for constructing executable 
+software from source. 
+
+## Why
+
+## Details
+
+### Goals
+
+Any workable solution must satisfy the following goals 
+
+ 1. The build system must be able to create repeatable builds, from source, of
+    any revision of the Teleport source at any time. 
+
+ 2. The same process (and its resulting images) should be used by all parts 
```

Comment:
Is it true that the same image should be used by all parts of the cycle? I often wonder things like:

- should I need a Firestore emulator just to run `make grpc`?
- should CI need a protobuf compiler when the generated code is already in source control?

It seems there are different tools required at different parts of the development cycle, and re-using the same image where we don't need to leads to bloat and potentially increased attack surface area.

Reply:
Perhaps I should be a bit more specific, and specify that they should be used for all _builds_, not _all parts of the dev cycle_.

Reply:
To chime in for my new to teleport friends: a plethora of smaller buildboxes for each different step will be more confusing than a monolithic buildbox. I appropriate the well scoped nature of micro-build images, but I think it would be a bigger tooling ask to do them well.

I could be convinced either way if someone else is passionate about the topic.

Reply:
Yeah, I appreciate the idea - and our buildbox could _definitely_ do with a trim, but the effort in splitting the buildbox up and maintaining the different images feels like it would be prohibitive.

Nothing is stopping us doing it in the future, though.